---
title: "tingverse_vignette"
author: "Cole Nawrocki"
date: "2023-07-21"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tingverse)
library(tidyverse)
library(org.Hs.eg.db)
library(clusterProfiler)
library(parallel)
library(ggrepel)
library(knitr)
```

```
library(tingverse)
library(tidyverse)
library(org.Hs.eg.db)
library(clusterProfiler)
library(parallel)
library(ggrepel)
library(knitr)
```

## Read in Example Data 
This is old data from a Nanostring GeoMx experiment. `geom` is a normalized counts table. `metadata` is the meta data for each AOI in the experiment. scRNA-seq data could be used as well. You'll need to use your own data.

```{r message=FALSE, warning=FALSE}
geom <- read.delim("/Users/cnawrocki/Library/CloudStorage/OneDrive-UniversityofVirginia/Ting Lab/Summer 2022/COLE/HCC/GeoMx/Data/qNormCounts.tsv")
metadata <- read.csv("/Users/cnawrocki/Library/CloudStorage/OneDrive-UniversityofVirginia/Ting Lab/Summer 2022/COLE/HCC/GeoMx/Data/meta_small.csv")

# Some basic manipulations to make the data files compatible. 
colnames(geom) <- substr(colnames(geom),14,23)
metadata$Sample_ID_short<-gsub('-','.',metadata$Sample_ID_short)
geom <- geom[,colnames(geom) %in% metadata$Sample_ID_short]
rownames(metadata)<-metadata$Sample_ID_short
metadata <- metadata[metadata$Sample_ID_short %in% colnames(geom),]
```

## Differential Expression
Use the `differential.expression()` function first. Then, use the `volcano.plot()` function to visualize the results. FUse `?differential.expression` and `?volcano.plot` to view documentation.

```{r message=FALSE, warning=FALSE}
DEA <- differential.expression(counts.table = geom,meta.data = metadata,meta.feature = "Tissue_Type",value.1 = "Tumor",value.2 = "Vessel",test = "wilcox")

volcano.plot(DEA,color.vals = c("green3","grey","red3"))
```

## GSEA
`?easy.gsea` will show documentation.

```{r}
GSEA_obj <- easy.gsea(DEA,order.stat = "log2.FC",ontology = "BP",organism = "human")

# Viewing the results
kable(head(as.data.frame(GSEA_obj)),format = 'markdown')
```

#### Looking at a Specific Gene Set
We can subset for only gene sets with 50 or more genes in them. Then, we can output the GSEA plot for the gene set with the highest enrichment score. `gseaplot` is a `ClusterProfiler` function.

```{r fig.height=12, fig.width=12, message=FALSE, warning=FALSE}
results_oi <- GSEA_obj@result[order(GSEA_obj@result$enrichmentScore,decreasing = T) & GSEA_obj@result$setSize > 50,]

gseaplot(GSEA_obj,geneSetID = results_oi$ID[1])
```

#### Dotplot Based on Our Results
`dotplot` is a `ClusterProfiler` function. Here, we will look at the top 50 gene sets and separate based on if they are activated or deactivated.

```{r fig.height=12.5, fig.width=12, message=FALSE, warning=FALSE}
dotplot(GSEA_obj,showCategory=50,split=".sign") + facet_grid(.~.sign) + theme(axis.text.y = element_text(size = 3))
```

#### Heatplot Based on Our Results
`heatplot` is a `ClusterProfiler` function. Here, we will look at the top 50 gene sets and visualize the LFC for each gene.

```{r fig.height=12, fig.width=12, message=FALSE, warning=FALSE}
heatplot(GSEA_obj,showCategory = 50,foldChange=GSEA_obj@geneList) + theme(axis.text.y = element_text(size = 4))
```

## Enrichment Analysis and Barplot
If we do not want to look at GSEA results, we can utilize this tool. We can look at all genes with a p.adj that is significant, run the analysis on them, then visualize the top 30 gene sets.

```{r fig.height=12, fig.width=12, message=FALSE, warning=FALSE}
genes_oi <- rownames(DEA[DEA$p.adj < 0.05,])
EA_obj <- enrichGO(gene = genes_oi,OrgDb = "org.Hs.eg.db",keyType = "SYMBOL",ont = "BP")
barplot(EA_obj, showCategory = 30) + theme(axis.text.y = element_text(size = 4))
```
